src/Type/Enum/EnumCaseObjectType.php

```diff
@@ -79,7 +79,7 @@ public function isSuperTypeOf(Type $type): TrinaryLogic
 			return $type->isSubTypeOf($this);
 		}
 
-		$parent = new parent($this->getClassName(), $this->getSubtractedType(), $this->getClassReflection());
+		$parent = new parent($this->getClassName(), $this->getSubtractedType());
```

Comment:
The problem with this is loss of precision. ObjectType often has ClassReflection given be the current scope. It makes analysis more precise if there are multiple classes with the same name - it's going to use reflection from the current file for real.

I have an idea: use classReflection if it's been set but don't fetch it from ReflectionProvider.

Reply:
it seems we are missing test-coverage for the case you mentioned.

just added a new method to get but not create the class-reflection of the base-class